companion/src/apppreferencesdialog.cpp

```diff
@@ -137,13 +134,11 @@ void AppPreferencesDialog::initSettings()
     ui->snapshotPath->setDisabled(true);
     ui->snapshotPathButton->setDisabled(true);
   }
-#if defined(ALLOW_NIGHTLY_BUILDS)
-  ui->useCompanionNightlyBuilds->setChecked(g.useCompanionNightlyBuilds());
-  ui->useFirmwareNightlyBuilds->setChecked(g.useFirmwareNightlyBuilds());
-#else
-  ui->useCompanionNightlyBuilds->hide();
-  ui->useFirmwareNightlyBuilds->hide();
+#if !defined(ALLOW_NIGHTLY_BUILDS)
+  // TODO should we gray out nightly builds here?
 #endif
+  ui->firmwareBranch->setCurrentIndex(g.firmwareBranch());
+  ui->companionBranch->setCurrentIndex(g.companionBranch());
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1135)